### PR TITLE
Removing calls to the internal API - CordaFutureImpl

### PR DESCRIPTION
**Pull request by mkit**

### CODE REVIEW

File: samples/attachment-demo/src/integration-test/kotlin/net/corda/attachmentdemo/AttachmentDemoTest.kt [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
             val demoUser = listOf(User("demo", "demo", setOf(startFlowPermission<AttachmentDemoFlow>())))
-            val (nodeA, nodeB) = listOf(
-                    startNode(DUMMY_BANK_A.name, rpcUsers = demoUser),
-                    startNode(DUMMY_BANK_B.name, rpcUsers = demoUser),
-                    startNode(DUMMY_NOTARY.name, setOf(ServiceInfo(SimpleNotaryService.type)))
-            ).transpose().getOrThrow()
+            val nodeAFuture = startNode(DUMMY_BANK_A.name, rpcUsers = demoUser).toCompletableFuture()
+            val nodeBFuture = startNode(DUMMY_BANK_B.name, rpcUsers = demoUser).toCompletableFuture()
+            val notaryFuture = startNode(DUMMY_NOTARY.name, setOf(ServiceInfo(SimpleNotaryService.type))).toCompletableFuture()
+            allOf(nodeAFuture, nodeBFuture, notaryFuture).getOrThrow()
```
Comment (andr3ej) [inline samples/attachment-demo/src/integration-test/kotlin/net/corda/attachmentdemo/AttachmentDemoTest.kt]: this allOf is redundant, as we getOrThrow them all on the next line anyway. and if we don't need allOf we don't need toCompletableFuture. and there are a few other places where you can make the same improvements

File: samples/network-visualiser/src/main/kotlin/net/corda/netmap/simulation/IRSSimulation.kt [hunk 2]
```diff
@@ -42,46 +42,58 @@ class IRSSimulation(networkSendManuallyPumped: Boolean, runAsync: Boolean, laten
...
+                }
             }
-        }, {})
-        return future
+            future
+        }
     }
 
-    private fun doNextFixing(i: Int, j: Int): CordaFuture<Unit>? {
+    private fun doNextFixing(i: Int, j: Int): CompletableFuture<*>? {
```
Comment (andr3ej) [inline samples/network-visualiser/src/main/kotlin/net/corda/netmap/simulation/IRSSimulation.kt]: it's generally bad practice to return a wildcard, please use CompletableFuture\<Void>

File: samples/network-visualiser/src/main/kotlin/net/corda/netmap/simulation/IRSSimulation.kt [hunk 3]
```diff
@@ -141,9 +153,9 @@ class IRSSimulation(networkSendManuallyPumped: Boolean, runAsync: Boolean, laten
         val acceptDealFlows: Observable<AcceptDealFlow> = node2.registerInitiatedFlow(AcceptDealFlow::class.java)
 
         @Suppress("UNCHECKED_CAST")
-       val acceptorTxFuture = acceptDealFlows.toFuture().flatMap {
-            (it.stateMachine as FlowStateMachine<SignedTransaction>).resultFuture
-        }
+       val acceptorTxFuture = acceptDealFlows.toFuture().toCompletableFuture().thenCompose({
```
Comment (andr3ej) [inline samples/network-visualiser/src/main/kotlin/net/corda/netmap/simulation/IRSSimulation.kt]: might as well fix the indent, also in kotlin you can pass the code block into thenCompose without parentheses

File: samples/network-visualiser/src/main/kotlin/net/corda/netmap/simulation/IRSSimulation.kt [hunk 4]
```diff
@@ -154,7 +166,7 @@ class IRSSimulation(networkSendManuallyPumped: Boolean, runAsync: Boolean, laten
                 node1.services.legalIdentityKey)
         val instigatorTxFuture = node1.services.startFlow(instigator).resultFuture
 
-        return listOf(instigatorTxFuture, acceptorTxFuture).transpose().flatMap { instigatorTxFuture }
+        return allOf(instigatorTxFuture.toCompletableFuture(), acceptorTxFuture).thenCompose({ instigatorTxFuture.toCompletableFuture() })
```
Comment (andr3ej) [inline samples/network-visualiser/src/main/kotlin/net/corda/netmap/simulation/IRSSimulation.kt]: redundant parentheses again

File: samples/network-visualiser/src/main/kotlin/net/corda/netmap/simulation/Simulation.kt [hunk 5]
```diff
@@ -261,21 +260,19 @@ abstract class Simulation(val networkSendManuallyPumped: Boolean,
...
 
-    val networkInitialisationFinished = mockNet.nodes.map { it.nodeReadyFuture }.transpose()
+    val networkInitialisationFinished = allOf(*mockNet.nodes.map { it.nodeReadyFuture.toCompletableFuture() }.toTypedArray())
 
-    fun start(): CordaFuture<Unit> {
+    fun start(): Future<Unit> {
         mockNet.startNodes()
         // Wait for all the nodes to have finished registering with the network map service.
-        return networkInitialisationFinished.flatMap { startMainSimulation() }
+        return networkInitialisationFinished.thenCompose({startMainSimulation()})
```
Comment (andr3ej) [inline samples/network-visualiser/src/main/kotlin/net/corda/netmap/simulation/Simulation.kt]: please use formatter on new code, also redundant parentheses

File: samples/notary-demo/src/main/kotlin/net/corda/notarydemo/Notarise.kt [hunk 6]
```diff
@@ -53,9 +51,12 @@ private class NotaryDemoClientApi(val rpc: CordaRPCOps) {
      * as it consumes the original asset and creates a copy with the new owner as its output.
      */
     private fun buildTransactions(count: Int): List<SignedTransaction> {
-        return (1..count).map {
-            rpc.startFlow(::DummyIssueAndMove, notary, counterpartyNode.legalIdentity, it).returnValue
-        }.transpose().getOrThrow()
+
```
Comment (andr3ej) [inline samples/notary-demo/src/main/kotlin/net/corda/notarydemo/Notarise.kt]: remove blank line

File: samples/simm-valuation-demo/src/main/kotlin/net/corda/vega/analytics/OGUtils.kt [hunk 7]
```diff
@@ -1,13 +1,7 @@
...
 
 fun compareIMTriples(a: InitialMarginTriple, b: InitialMarginTriple): Boolean {
-    if (a.first is Double && a.second is Double && a.third is Double &&
-            b.first is Double && b.second is Double && b.third is Double) {
-        if (withinTolerance(a.first, b.first) && withinTolerance(a.second, b.second) && withinTolerance(a.third, b.third)) {
-            return true
-        }
-    }
-    return false
+    return withinTolerance(a.first, b.first) && withinTolerance(a.second, b.second) && withinTolerance(a.third, b.third)
```
Comment (andr3ej) [inline samples/simm-valuation-demo/src/main/kotlin/net/corda/vega/analytics/OGUtils.kt]: i'll allow this, but usually refactoring should be related to the business task

**Comment by mkit** [comment]
Review comments addressed.

File: samples/attachment-demo/src/integration-test/kotlin/net/corda/attachmentdemo/AttachmentDemoTest.kt [hunk 8]
```diff
@@ -1,37 +1,34 @@
...
         driver(dsl = {
             val demoUser = listOf(User("demo", "demo", setOf(startFlowPermission<AttachmentDemoFlow>())))
-            val (nodeA, nodeB) = listOf(
-                    startNode(DUMMY_BANK_A.name, rpcUsers = demoUser),
-                    startNode(DUMMY_BANK_B.name, rpcUsers = demoUser),
-                    startNode(DUMMY_NOTARY.name, setOf(ServiceInfo(SimpleNotaryService.type)))
-            ).transpose().getOrThrow()
+            startNode(DUMMY_NOTARY.name, setOf(ServiceInfo(SimpleNotaryService.type))).getOrThrow()
+            val nodeA = startNode(DUMMY_BANK_A.name, rpcUsers = demoUser).getOrThrow()
+            val nodeB = startNode(DUMMY_BANK_B.name, rpcUsers = demoUser).getOrThrow()
```
Comment (andr3ej) [inline samples/attachment-demo/src/integration-test/kotlin/net/corda/attachmentdemo/AttachmentDemoTest.kt]: new problem: previously the nodes were started in parallel, now you're waiting for one to finish before starting the next one

File: samples/notary-demo/src/main/kotlin/net/corda/notarydemo/Notarise.kt [hunk 9]
```diff
@@ -54,8 +51,8 @@ private class NotaryDemoClientApi(val rpc: CordaRPCOps) {
      */
     private fun buildTransactions(count: Int): List<SignedTransaction> {
         return (1..count).map {
-            rpc.startFlow(::DummyIssueAndMove, notary, counterpartyNode.legalIdentity, it).returnValue
-        }.transpose().getOrThrow()
+            rpc.startFlow(::DummyIssueAndMove, notary, counterpartyNode.legalIdentity, it).returnValue.getOrThrow()
```
Comment (andr3ej) [inline samples/notary-demo/src/main/kotlin/net/corda/notarydemo/Notarise.kt]: similarly, these flows used to be launched in parallel

**Comment by mkit** [comment]
Good point about nodes not starting in parallel. Thank you for that. I have addressed this.

File: samples/simm-valuation-demo/src/test/kotlin/net/corda/vega/Main.kt [hunk 10]
```diff
@@ -17,12 +16,11 @@ import net.corda.testing.driver.driver
...
-        val (nodeA, nodeB, nodeC) = listOf(
-                startNode(providedName = DUMMY_BANK_A.name),
-                startNode(providedName = DUMMY_BANK_B.name),
-                startNode(providedName = DUMMY_BANK_C.name)
-        ).transpose().getOrThrow()
+        val notaryFuture = startNode(providedName = DUMMY_NOTARY.name, advertisedServices = setOf(ServiceInfo(SimpleNotaryService.type)))
+        val nodeAFuture = startNode(providedName = DUMMY_BANK_A.name)
+        val nodeBFuture = startNode(providedName = DUMMY_BANK_B.name)
+        val nodeCFuture = startNode(providedName = DUMMY_BANK_C.name)
+        val (nodeA, nodeB, nodeC) = listOf(nodeAFuture, nodeBFuture, nodeCFuture, notaryFuture).map { it.getOrThrow() }
```
Comment (andr3ej) [inline samples/simm-valuation-demo/src/test/kotlin/net/corda/vega/Main.kt]: the old code didn't wait for the notary, but it probably should have done

**Pull request merged**